plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServerSettingsViewModel.java

```diff
@@ -338,6 +450,90 @@ private String getProfileId() {
 		return OpenShiftServerBehaviour.PROFILE_OPENSHIFT3;
 	}
 
+	private String getRoute(boolean isDefaultRoute, IRoute route) {
+		if (!isDefaultRoute || route == null) {
+			return null;
+		}
+		return route.getURL();
+	}
+
+	public boolean isSelectDefaultRoute() {
+		return selectDefaultRoute;
+	}
+
+	public void setSelectDefaultRoute(boolean selectDefaultRoute) {
+		update(getConnection(), getConnections(), 
+				this.deployProject, this.projects, 
+				this.sourcePath, podPath, this.imageStreamTagsMap, 
+				getService(), getServiceItems(),
+				this.route, selectDefaultRoute, this.routesByProject);
+	}
+
+	protected void setRoutes(Map<IProject, List<IRoute>> routesByProject) {
+		update(getConnection(), getConnections(), 
+				this.deployProject, this.projects, 
+				this.sourcePath, this.podPath, this.imageStreamTagsMap, 
+				getService(), getServiceItems(),
+				this.route, this.selectDefaultRoute, routesByProject);
+	}
+	
+	public List<IRoute> getRoutes() {
+		if (getService() == null) {
+			return Collections.emptyList();
+		}
+		return getAllRoutes(getService());
```

Comment:
This now returns all routes for the project. Should not they be filtered by areRelated(IRoute route, IService service) ?